server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/ServerApiHelper.java

```diff
@@ -207,6 +207,75 @@ private static String tryParseAsJsonError(HttpClient.Response response) {
     progress.setProgressAndCheckCancel("Page " + page, loaded.get() / (float) paging.getTotal());
   }
 
+  public HttpClient.AsyncRequest getEventStream(String path, Consumer<RawEvent> rawEventConsumer) {
+    var eventBuffer = new EventBuffer();
+    return client.getEventStream(buildEndpointUrl(path),
+      rawData -> eventBuffer.append(rawData)
+        .getCompleteEvents()
+        .forEach(rawEvent -> rawEventConsumer.accept(parseEvent(rawEvent))))
+      .map(futureResponse -> futureResponse.thenApply(response -> {
+        if (!response.isSuccessful()) {
+          throw handleError(response);
+        }
+        return response;
+      }));
+  }
+
+  public static class RawEvent {
+    private final String type;
+    private final String data;
+
+    public RawEvent(String type, String data) {
+      this.type = type;
+      this.data = data;
+    }
+
+    public String getType() {
+      return type;
+    }
+
+    public String getData() {
+      return data;
+    }
+  }
+
+  private static class EventBuffer {
+    private final StringBuilder buffer = new StringBuilder();
+
+    private EventBuffer append(String data) {
+      buffer.append(data);
+      return this;
+    }
+
+    private List<String> getCompleteEvents() {
+      List<String> completeEvents = new ArrayList<>();
+      int firstEventEndIndex;
+      do {
+        firstEventEndIndex = buffer.indexOf("\r\n");
+        if (firstEventEndIndex == -1) {
+          break;
+        }
+        completeEvents.add(buffer.substring(0, firstEventEndIndex));
+        buffer.delete(0, firstEventEndIndex + 2);
+      } while (true);
+      return completeEvents;
+    }
+  }
+
+  private static RawEvent parseEvent(String eventPayload) {
+    var fields = List.of(eventPayload.split("\\n"));
+    var type = "";
+    var data = new StringBuilder();
+    for (String field : fields) {
+      if (field.startsWith("event: ")) {
+        type = field.substring(7);
```

Comment:
```suggestion
        type = field.substring("event: ".length());
```
or better use a constant